### DISCUSSION TITLE
result without top-level paragraph for single lines?

### DISCUSSION THREAD

**Discussion by artificiel** [facelessuser/pymdown-extensions#1890] [Q&A] [answered]
I am confused by the interpretation of paragraphs in pymdown. the front page has the example:

text = "A link https://google.com" which (thanks to magic link) gives 

`<p>A link <a href="https://google.com">https://google.com</a></p>`

why is the output surrounded by `<p/>`? the markdown spec says: _A paragraph is simply one or more consecutive lines of text, separated by one or more blank lines._ however there is zero blank line in the input above -- where does the paragraph come from?

according to the spec the expected output would be `A link <a href="https://google.com">https://google.com</a>`

**Comment by facelessuser** [comment, marked as answer]
> according to the spec the expected output would be `A link <a href="https://google.com">https://google.com</a>`

No, according the Markdown spec, a paragraph gets wrapped in `<p></p>`.  https://daringfireball.net/projects/markdown/syntax#p

This isn't a pymdown question and has nothing to do with MagicLink either. This is just how Markdown works.

This site shows multiple Markdown parsers doing the same thing with a single sentence: https://babelmark.github.io/?text=Some+text.
